### ISSUE TITLE
Azure SD Config targets entire subscription and not only the specified resource_group

### ISSUE THREAD

**Issue by shanoor** [VictoriaMetrics/VictoriaMetrics#7630]
### Describe the bug

I'm trying to scrape metrics from Azure VM using `azure_sd_configs` but I noticed the `resource_group` config is ignored and victoria-agent tries to scrape VM in the entire subscription and ends up with hundreds of scrape target error. The same config works as expected with Prometheus.

### To Reproduce

- Have two or more Azure resource groups containing virtual machines
- Use a config like below to target one resource group :
    ```
    extraScrapeConfigs:
      - job_name: scrape-metrics-vm
        azure_sd_configs:
          - authentication_method: ManagedIdentity
            environment: AzurePublicCloud
            subscription_id: xxxxxxxxxxxxxxxxxxxxxxxxxx
            resource_group: RG001
            port: 9182
        relabel_configs:
          - source_labels: [__meta_azure_machine_location]
            target_label: location
          - source_labels: [__meta_azure_machine_name]
            target_label: vm_name
          - source_labels: [__meta_azure_machine_os_type]
            target_label: os_type
          - source_labels: [__meta_azure_machine_private_ip]
            target_label: private_ip
          - source_labels: [__meta_azure_machine_resource_group]
            target_label: resource_group
          - source_labels: [__meta_azure_subscription_id]
            target_label: subscription_id
    ```
- you can check in victoria-agent's ui that all VM of the subscription are targeted not just the ones in the specified resource_group.

### Version

vmagent-20241115-224452-tags-v1.106.1-0-g4b74baf696

**Comment by jiekun** [comment]
Hello. I'm trying to investigate this but still have trouble using the Azure platform to obtain my own client_id, secret and etc...I will try to figure out. But in the mean time, it woule be nice if you have a valid authentication info for me to test. Thanks!

**Comment by jiekun** [comment]
Hi. @shanoor Sorry for the delay. I just created a pull request to fix the `resource_group` filter for Azure service discovery on **virtual machine scale sets**. Previously, this filter **did not apply to virtual machine scale sets**, causing all virtual machines to be discovered. 

It would be great if you could test the following image (or [build it]( https://docs.victoriametrics.com/#how-to-build-from-sources) from `feature/azure-sd-resource-group` branch): 
```
victoriametrics/vmagent:heads-feature-azure-sd-resource-group-0-gaeab56ad9
```

**Comment by PaulPowershell** [comment]
hi @jiekun, i just test with image heads-feature-azure-sd-resource-group-0-gaeab56ad9 and it's OK now
thx for your answer

**Comment by shanoor** [comment]
Hi @jiekun, many thanks for the quick fix, we confirmed that it now works as intended 👍

**Comment by jiekun** [comment]
Cool. Please try to update to the new version (probably `v1.106.2`) when it's released. The test image is based on our `master` branch and may contain other unreleased features **and bugs**.

**Comment by f41gh7** [comment]
The bugfix was included into [v1.107.0](https://docs.victoriametrics.com/changelog/#v11070), [v1.102.8](https://docs.victoriametrics.com/changelog/#v11028) and [v1.97.13](https://docs.victoriametrics.com/changelog/#v19713)
